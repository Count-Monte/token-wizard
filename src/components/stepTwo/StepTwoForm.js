import React from 'react'
import { FormSpy } from 'react-final-form'
import { TokenName } from '../Common/TokenName'
import { TokenTicker } from '../Common/TokenTicker'
import { TokenDecimals } from '../Common/TokenDecimals'

const errorStyle = {
  color: 'red',
  fontWeight: 'bold',
  fontSize: '12px',
  width: '100%',
  height: '10px',
}

export const StepTwoForm = ({ id, handleSubmit, disableDecimals, updateTokenStore }) => {
  return (
    <form id={id} onSubmit={handleSubmit}>
      <div className="hidden">
        <TokenName errorStyle={errorStyle}/>
        <TokenTicker errorStyle={errorStyle}/>
        <TokenDecimals disabled={disableDecimals} errorStyle={errorStyle}/>
      </div>
      <FormSpy
        onChange={({ values }) => {
          updateTokenStore('name', values.name)
          updateTokenStore('ticker', values.ticker)
          updateTokenStore('decimals', values.decimals)
        }}
      />
    </form>
  )
}
