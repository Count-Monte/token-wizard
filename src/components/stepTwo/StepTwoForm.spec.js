import React from 'react'
import { Form } from 'react-final-form'
import { StepTwoForm } from './StepTwoForm'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'

configure({ adapter: new Adapter() })

describe('StepTwoForm', () => {
  it('Should render the component', () => {
    const component = renderer.create(
      <Form
        onSubmit={jest.fn()}
        initialValues={{}}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={jest.fn()}
        id="tokenData"
        tokens={[]}
      />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should call onSubmit if all values are valid', () => {
    const onSubmit = jest.fn()
    const wrapper = mount(
      <Form
        onSubmit={onSubmit}
        initialValues={{
          name: 'This is a valid name',
          ticker: 'TTK',
          decimals: '14'
        }}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={jest.fn()}
        id="tokenData"
        tokens={[]}
      />
    )

    wrapper.simulate('submit')

    expect(onSubmit).toHaveBeenCalled()
  })

  it('Should not call onSubmit if not all values are valid', () => {
    const onSubmit = jest.fn()
    const wrapper = mount(
      <Form
        onSubmit={onSubmit}
        initialValues={{
          name: 'This is a valid name',
          ticker: 'TTK',
          decimals: '24'
        }}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={jest.fn()}
        id="tokenData"
        tokens={[]}
      />
    )

    wrapper.simulate('submit')

    expect(onSubmit).toHaveBeenCalledTimes(0)
  })

  it('Should call updateTokenStore after changing name value', () => {
    const updateTokenStore = jest.fn()
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()}
        initialValues={{}}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={updateTokenStore}
        id="tokenData"
        tokens={[]}
      />
    )

    const decimals = wrapper.find('input[name="name"]')
    decimals.simulate('change', { target: { value: 'MyToken' } })

    expect(updateTokenStore).toHaveBeenCalled()
  })

  it('Should call updateTokenStore after changing ticker value', () => {
    const updateTokenStore = jest.fn()
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()}
        initialValues={{}}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={updateTokenStore}
        id="tokenData"
        tokens={[]}
      />
    )

    const decimals = wrapper.find('input[name="ticker"]')
    decimals.simulate('change', { target: { value: 'MyToken' } })

    expect(updateTokenStore).toHaveBeenCalled()
  })

  it('Should call updateTokenStore after changing decimal value', () => {
    const updateTokenStore = jest.fn()
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()}
        initialValues={{}}
        component={StepTwoForm}
        disableDecimals={false}
        updateTokenStore={updateTokenStore}
        id="tokenData"
        tokens={[]}
      />
    )

    const decimals = wrapper.find('input[name="decimals"]')
    decimals.simulate('change', { target: { value: 'MTK' } })

    expect(updateTokenStore).toHaveBeenCalled()
  })
})
