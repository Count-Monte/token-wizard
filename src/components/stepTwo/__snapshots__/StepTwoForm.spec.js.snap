// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepTwoForm Should render the component 1`] = `
<form
  id="tokenData"
  onSubmit={[Function]}
>
  <div
    className="hidden"
  >
    <div
      className="left"
    >
      <label
        className="label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        autoComplete="off"
        className="input"
        disabled={undefined}
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="description"
      >
        The name of your token. Will be used by Etherscan and other tokenbrowsers. Be afraid of trademarks.
      </p>
      <span>
        <p
          className="error"
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </span>
    </div>
    <div
      className="right"
    >
      <label
        className="label"
        htmlFor="ticker"
      >
        Ticker
      </label>
      <input
        autoComplete="off"
        className="input"
        disabled={undefined}
        id="ticker"
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="description"
      >
        The five letter ticker for your token. There are 11,881,376 combinations for 26 english letters.
                  Be hurry.
      </p>
      <span>
        <p
          className="error"
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </span>
    </div>
    <div
      className="left"
    >
      <label
        className="label"
        htmlFor="decimals"
      >
        Decimals
      </label>
      <input
        autoComplete="off"
        className="input"
        disabled={false}
        id="decimals"
        name="decimals"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="description"
      >
        Refers to how divisible a token can be, from 0 (not at all divisible) to 18 (pretty much continuous).
      </p>
      <span>
        <p
          className="error"
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </span>
    </div>
  </div>
  <div
    className="reserved-tokens-title"
  >
    <p
      className="title"
    >
      Reserved tokens
    </p>
  </div>
  <div
    className="reserved-tokens-container"
  >
    <div
      className="reserved-tokens-input-container"
    >
      <div
        className="reserved-tokens-input-container-inner"
      >
        <div
          className="reserved-tokens-input-property reserved-tokens-input-property-left"
        >
          <label
            className="label"
          >
            Address
          </label>
          <input
            className="input"
            disabled={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onKeyPress={undefined}
            onPaste={undefined}
            style={undefined}
            type="text"
            value=""
          />
          <p
            className="description"
          >
            Address where to send reserved tokens.
          </p>
          <p
            style={
              Object {
                "color": "red",
                "fontSize": "12px",
                "fontWeight": "bold",
                "height": "10px",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="reserved-tokens-input-property reserved-tokens-input-property-middle"
        >
          <label
            className="label"
          >
            Dimension
          </label>
          <div
            className="radios-inline"
          >
            <label
              className="radio-inline"
            >
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="tokens"
              />
              <span
                className="title"
              >
                tokens
              </span>
            </label>
            <label
              className="radio-inline"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="percentage"
              />
              <span
                className="title"
              >
                percentage
              </span>
            </label>
          </div>
          <p
            className="description"
          >
            Fixed amount or % of crowdsaled tokens. Will be deposited to the account after finalization of the crowdsale.
          </p>
        </div>
        <div
          className="reserved-tokens-input-property reserved-tokens-input-property-right"
        >
          <label
            className="label"
          >
            Value
          </label>
          <input
            className="input"
            disabled={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            style={undefined}
            type="number"
            value=""
          />
          <p
            className="description"
          >
            Value in tokens. Don't forget to click + button for each reserved token.
          </p>
          <p
            style={
              Object {
                "color": "red",
                "fontSize": "12px",
                "fontWeight": "bold",
                "height": "10px",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="plus-button-container"
      >
        <div
          className="button button_fill button_fill_plus"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="button-container"
  >
    <a
      className="button button_fill"
      onClick={[Function]}
    >
      Continue
    </a>
  </div>
</form>
`;
